src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -587,6 +589,49 @@ private static Expression BindParameterFromValue(ParameterInfo parameter, Expres
             return argument;
         }
 
+        private static MethodCallExpression CreateTryParseCall(MethodInfo tryParseMethod, ParameterInfo parameter, Expression parsedValue)
+        {
+            // Before call the TryParse Method, we should know the exact param to inject
+            Type type = Nullable.GetUnderlyingType(parameter.ParameterType) ?? parameter.ParameterType;
+            var useNumberStyles = TryParseMethodCache.UseTryParseWithNumberStyleOption(type);
```

Comment:
Why not rename this to `GetTryParseNumberStyle` just have this return the NumberStyle or null? That way you don't have to call `SetRightNumberStyles` later and repeat this exact same check.

Reply:
Maybe we can integrate this in **`TryGetNumberStylesTryGetMethod(Type type, out MethodInfo? method, out NumberStyles? numberStyles)`**. with that, we call this only and get the **`TryParse`** method and the right **`NumberStyle`**. 🙂